components/table/TableRow.js

```diff
@@ -60,6 +60,9 @@ const factory = (Checkbox) => {
     renderInput (key, value) {
       const index = this.props.index;
       const inputType = utils.inputTypeForPrototype(this.props.model[key].type);
+      if (this.props.model[key].render) {
+        return this.props.model[key].render(this.props.data);
```

Comment:
No, it calls a render function supplied in the data model, the test for props.model[key].render is checking for the existence of a render function in the model, if it finds one then it's used to render the table cell and further rending is prevented by early return.

I'm implementing code from @joehua87 but I think his intent was reasonably clear and it seems to work pretty well.

This is in answer to #268 which describes the problem and this solution more detail but essentially, the model becomes:

`const model={ someColumn: {type: String, render: function(data){ return (MyDropDown(data)) } } };`
Where MyDropDown is defined elsewhere in scope:
```
const dropDownData = {...};
const MyDropDown = function(data) { return (<Dropdown
        auto
        label={'Choose data item'}
        onChange={this.handleMDDChange}
        source={dropDownData}
        value={this.state.source.dropDownValue} />
```)};

This lets the dev embed arbitrary controls into the table cell and use the control's onChange to keep them synchronised with the table state.